packages/hardhat-chai-matchers/src/internal/utils.ts

```diff
@@ -10,3 +14,14 @@ export function assertIsNotNull<T>(
     );
   }
 }
+
+export function preventAsyncMatcherChaining(
+  context: object,
+  matcherName: string,
+  chaiUtils: Chai.ChaiUtils
+) {
+  if (chaiUtils.flag(context, ASYNC_MATCHER_CALLED) === true) {
+    throw new HardhatChaiMatchersNonChainableMatcherError(matcherName);
+  }
+  chaiUtils.flag(context, ASYNC_MATCHER_CALLED, true);
```

Comment:
Random idea, we could use the matcher name as the flag value:

```suggestion
  const previousMatcherName = chaiUtils.flag(context, ASYNC_MATCHER_CALLED);
  if (previousMatcherName !== undefined) {
    throw new HardhatChaiMatchersNonChainableMatcherError(matcherName, previousMatcherName);
  }
  chaiUtils.flag(context, ASYNC_MATCHER_CALLED, matcherName);
```

and then change the error message to say something like `${matcherName} was called after ${previousMatcherName}`. Wdyt?

Reply:
I like it 👍